### PR TITLE
Light Client benchmarks

### PR DESCRIPTION
**Pull request by fbielejec**
This PR adds the benchmarking to the LC pallet. The baseline numbers (`weights.rs`) are generated are on my machine, we should probably follow up with a task that runs them on the same hardware our validators are running on.

### CODE REVIEW

**Comment by deuszx** [comment]
Can we extract the unrelated changes to separate PRs? For example:
* addition of the benchmark command could be merged directly to `main`, correct?
* adding an RPC server for the pallet could be a separate PR
* etc.

So that this one contains only the benchmark code.

File: pallets/tendermint-light-client/src/lib.rs [hunk 1]
```diff
@@ -57,6 +61,9 @@ pub mod pallet {
 
         /// time provider type, used to gauge whether blocks are within the trusting period
         type TimeProvider: UnixTime;
+
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
```
Comment (deuszx) [inline pallets/tendermint-light-client/src/lib.rs:66]: `mock.rs` does not compile for me b/c it's missing this type now.

**Comment by fbielejec** [comment]
> Can we extract the unrelated changes to separate PRs? For example:
> 
> * addition of the benchmark command could be merged directly to `main`, correct?
> * adding an RPC server for the pallet could be a separate PR
> * etc.
> 
> So that this one contains only the benchmark code.

AFAIK all the changes in this PR are related to the benchmarks only.

File: pallets/tendermint-light-client/src/benchmarks.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            options,
+            initial_block
+        ));
+
+        let caller: T::AccountId = whitelisted_caller();
+        let untrusted_block = blocks.pop ().unwrap ();
+
+    }: update_client(RawOrigin::Signed(caller.clone()), untrusted_block.clone ())
+
+        verify {
```
Comment (Marcin-Radecki) [inline pallets/tendermint-light-client/src/benchmarks.rs:76]: I can see that `update_client` has a lot of corner cases, I assume we test here only sunny case scenario which is the most complex from runtime pov?

**Review by Marcin-Radecki — CHANGES_REQUESTED** [review]
Most of the comments are questions and docs requests.

File: pallets/tendermint-light-client/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Tendermint LC pallet
+
+## Run benchmarks
+
+```rust
+cargo build --release --features runtime-benchmarks
+
+./target/release/aleph-node benchmark --chain docker/data/chainspec.json --extrinsic='*' --pallet=pallet-tendermint-light-client s --template=./.maintain/pallet-weight-template.hbs --output ./pallets/tendermint-light-client/src/weights.rs
```
[the remark this answers is not in the document]
Comment (fbielejec) [reply pallets/tendermint-light-client/README.md:8]: Done

**Comment by fbielejec** [comment]
@Marcin-Radecki  I fixed the most obvious points you raised, but I'm merging as-is to the upstream branch - no point in spending more time on it since it goes to the freezer for the foreseeable future :shrug:
